### PR TITLE
feat(footer): introducing micro variation

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)

#3426 

### Description

Introducing the Micro variation of Footer in Web Components. 

### Changelog

**New**

- micro variation for `<dds-footer>`

**Changed**

- `legal-nav` list styling refactored to only apply to child `<ul>` element

### CODE REVIEW

File: packages/web-components/src/components/footer/legal-nav.ts [hunk 1]
```diff
@@ -23,6 +24,22 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-legal-nav`)
 class DDSLegalNav extends StableSelectorMixin(LitElement) {
+  /**
+   * Size property used for applying classes
+   */
+  @property()
+  size?: string;
```
Comment (asudoh) [inline packages/web-components/src/components/footer/legal-nav.ts:31]: Could you make this an enum? Ref: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/footer/footer.ts#L48

File: packages/web-components/src/components/footer/locale-button.ts [hunk 2]
```diff
@@ -25,6 +25,12 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-locale-button`)
 class DDSLocaleButton extends StableSelectorMixin(FocusMixin(LitElement)) {
+  /**
+   * Size property to apply different styles.
+   */
+  @property()
+  size?: string;
```
Comment (asudoh) [inline packages/web-components/src/components/footer/locale-button.ts:32]: Could you make this an enum? Ref: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/footer/footer.ts#L48

**Review by asudoh — COMMENTED** [review]
Nice progress @IgnacioBecerra! Another question; Any backgrounds on the style overhaul? Thanks!

**Comment by IgnacioBecerra** [comment]
> Nice progress @IgnacioBecerra! Another question; Any backgrounds on the style overhaul? Thanks!

There was a [note](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/4547/files#diff-1760beac32d616b50844cd75c4572a080f967e642a9891ec9f8af4833e9ebc07L72-L73) left in the styles file that had been there for some time now, so I went ahead with it and did it.

**Comment by asudoh** [comment]
>> Nice progress @IgnacioBecerra! Another question; Any backgrounds on the style overhaul? Thanks!

> There was a [note](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/4547/files#diff-1760beac32d616b50844cd75c4572a080f967e642a9891ec9f8af4833e9ebc07L72-L73) left in the styles file that had been there for some time now, so I went ahead with it and decided to fix it.

Thanks @IgnacioBecerra for clarifying! Seems that the comment was made my me, and it means more of eliminating Web Components-specific style. Seems that Web Components-style is still there... Would you want to forego the style overhaul part for now, or would you want to go ahead and eliminate Web Components-specific style as part of this PR? Either way is fine to me.

File: packages/web-components/src/components/footer/legal-nav.ts [hunk 3]
```diff
@@ -17,6 +17,26 @@ import styles from './footer.scss';
...
+  /**
+   * Short size.
+   */
+  SHORT = 'short',
+
+  /**
+   * Micro size.
+   */
+  MICRO = 'micro',
+}
```
Comment (asudoh) [inline packages/web-components/src/components/footer/legal-nav.ts:38]: Sorry I was not clear enough; Would you update `footer.ts` to add the micro variant and import the num from `footer.ts` to here? Same for the other file.

**Comment by IgnacioBecerra** [comment]
@asudoh Let's forego the style overhaul for now, although they will have to be tweaked so they doesn't clash against the micro styles (bringing the the old styles back make it so they're also being applied to the micro styles). Will be pushing the fixes soon.

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @IgnacioBecerra for the update!

File: packages/web-components/src/components/footer/footer-composite.ts [hunk 4]
```diff
@@ -204,14 +204,25 @@ class DDSFooterComposite extends ModalRenderMixin(HybridRenderMixin(HostListener
...
+            `
+          : html``}
+        <dds-legal-nav size="${ifNonNull(size)}">
           ${legalLinks?.map(
             ({ title, url }) => html`
               <dds-legal-nav-item href="${ifNonNull(url)}">${title}</dds-legal-nav-item>
             `
           )}
           <dds-legal-nav-cookie-preferences-placeholder></dds-legal-nav-cookie-preferences-placeholder>
+          ${size === 'micro'
```
Comment (asudoh) [inline packages/web-components/src/components/footer/footer-composite.ts:219]:
```suggestion
          ${size === FOOTER_SIZE.MICRO
```

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx [hunk 5]
```diff
@@ -30,7 +30,8 @@ import '@carbon/ibmdotcom-web-components/es/components/footer/footer-container';
 <dds-footer-container></dds-footer-container>
 ```
 
-> 💡 You can get short variant of footer with `size="short"` attribute.
+> 💡 You can get size variants of footer with `size="short"` or `size="short"` attribute.
```
Comment (asudoh) [inline packages/web-components/src/components/footer/__stories__/README.stories.mdx:33]:
```suggestion
> 💡 You can get size variants of footer with `size="short"` or `size="micro"` attribute.
```

**Review by kennylam — APPROVED** [review]
Looks good thanks @IgnacioBecerra!

**Pull request merged**